projects/storefrontlib/src/lib/cms/services/component-mapper.service.ts

```diff
@@ -58,4 +73,59 @@ export class ComponentMapperService {
 
     return factoryEntry ? factoryEntry[0] : null;
   }
+
+  isWebComponent(typeCode: string): boolean {
+    return (this.getType(typeCode) || '').includes('#');
+  }
+
+  initWebComponent(
+    componentType: string,
+    renderer: Renderer2
+  ): Promise<string> {
+    return new Promise(resolve => {
```

Comment:
If anything goes wrong, maybe it would be also good to consider rejecting a promise?